Automattic/jetpack
# Publicize: update message box

~# Branched off from https://github.com/Automattic/jetpack/pull/21016~

This PR reimplement the textarea control of the publicize plugin, aka MessageBox. But also, it implements a big refactoring on related code:

* Create the `<MessageBox />` component.
* Use `<TextareaControl />` instead of a native element.
* Reimplement components using functions
* Add two hooks to make the implementation clearer and simpler. More solid.

Fixes #

#### Changes proposed in this Pull Request:

* Update publicize textarea.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


NO

#### Testing instructions:






* Apply these changes
* Confirm the visual changes

<img src="https://user-images.githubusercontent.com/77539/132829234-81c2cba4-ffbf-4fb1-9ba0-c21d03f87b5b.png" width="300px" />

* Confirm the Publicize is still working as expected. Nothing should change in terms of functionality, compared with prod.
